src/middlewared/middlewared/plugins/network.py

```diff
@@ -173,6 +173,60 @@ async def toggle_announcement(self, data):
 
             await self.middleware.call(f'service.{verb}', service_name)
 
+    @private
+    async def propagate_changes_to_remote(self, remote_actions, local_config, licensed):
+        """
+        On the internal HA VM used for integration tests, we're seeing that propagating db changes
+        that have been written to the active node are taking up to 4 seconds to propagate to the
+        standby node and be flushed to disk. This means we'll restart the dns "service" on the remote
+        node before the db changes have been flushed which has the perception that the db changes aren't
+        actually getting propagating...but they are we just have a race...because of an unfortunate design.
```

Comment:
`s/propagating/propagated` ?

Reply:
heh, I actually fixed that in my 2nd commit so I'm assuming looking at cached version 😄

Reply:
Hah... my bad 🤪